edx_adapt/tests/logic_tests.py

```diff
@@ -19,18 +20,19 @@ def _setup_course_in_edxadapt(client, **kwargs):
         payload = json.dumps({'skill_name': skill})
         client.post(base_api_path + '/{course_id}/skill'.format(**kwargs), data=payload, headers=kwargs['headers'])
     path_to_file = os.path.join(os.path.dirname(__file__), '../../data/BKT/problems.csv')
-    for row in [line.strip().split(',') for line in open(path_to_file).readlines()]:
-        problem = row[0]
-        skill = row[1]
-        url = 'http://edx-lms.raccoongang.com/courses/{course_id}/{pname}'.format(pname=problem, **kwargs)
-        pre = "Pre_a" in problem
-        post = "Post_a" in problem
-        payload = json.dumps({
-            'problem_name': problem, 'tutor_url': url, 'skills': [skill], 'pretest': pre, 'posttest': post
-        })
-        client.post(
-            base_api_path + '/{course_id}'.format(**kwargs), data=payload, headers=kwargs['headers']
-        )
+    with open(path_to_file) as file_csv:
+        for row in csv.reader(file_csv):
+            problem = row[0]
+            skill = row[1]
+            url = 'http://edx-lms.raccoongang.com/courses/{course_id}/{pname}'.format(pname=problem, **kwargs)
+            pre = "Pre_a" in problem
+            post = "Post_a" in problem
+            payload = json.dumps({
```

Comment:
No need to encode into json manually. [Requests can do this for you](http://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests).